### PR TITLE
Vulpkanin can Growl and Howl

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Slimes can squish, IPCs can buzz, ping, and tone yes's and no's....and now...

Vulpkanin can growl and howl.

Credit to Goonstation for the growling and howling sounds.

Howl has a massive 10 second cooldown, growl is just a standard one.

And yes, you can howl *or* growl at specific people in view.

### CODE REVIEW

**Comment by KasparoVy** [comment]
:+1: All the fucking way across the sky :X: :100:

**Comment by Tastyfish** [comment]
Now we need tajaran growls for parity.

**Comment by LittleBigKid2000** [comment]
inb4 controversy

**Comment by KasparoVy** [comment]
Record someone's angry ass cat

**Comment by tigercat2000** [comment]
snahflak label please

**Comment by KasparoVy** [comment]
Fuck yeeeeah

**Comment by Yukiona** [comment]
:+1:!

**Comment by NTSAM** [comment]
needs snoflek geybel

**Comment by tigercat2000** [comment]
@TheDZD is give snahflak label please :cat:

**Comment by LittleBigKid2000** [comment]
Why can't they bark too

**Comment by KasparoVy** [comment]
cuz goon doesnt have barking noises

File: code/modules/mob/living/carbon/human/emote.dm [hunk 1]
```diff
@@ -63,6 +73,40 @@
...
+					if(param == A.name)
+						M = A
+						break
+			if(!M)
+				param = null
+
+			if(param)
+				message = "<B>[src]</B> howls at [param]!"
+			else
+				message = "<B>[src]</B> howls!"
```
Comment (tigercat2000) [inline code/modules/mob/living/carbon/human/emote.dm]: awooooooooooooo

**Comment by Tastyfish** [comment]
Oddly, goon actually does have a barking noise.

**Comment by Spacemanspark** [comment]
*scream

**Comment by Draconic-saint** [comment]
Vox shrieking all over again, calling it now
my poor ears....

**Comment by tigercat2000** [comment]
@Tastyfish yeah but that's a pussy barking noise
you want something like [this](https://youtu.be/AD8AN_Mmq5c?t=93)

**Comment by KasparoVy** [comment]
Yeah, if it's anything like what @tigercat2000 posted it'd be in theme with the badass sounding growls

**Comment by Iamgoofball** [comment]
:-1:
nope nope nope nope

you know how the goons were super worried that their shit was gonna get used for creepy furry crap

**this is exactly the sort of thing they were worried about**

**Comment by scrubmcnoob** [comment]
goof don't be xenophobic over a 2d spessman game.
@Fox-McCloud but what about the tangerines!!!?? They are like wolf guys or something, gotta let them do this as well!

**Comment by Iamgoofball** [comment]
furry isn't a race

**Comment by NTSAM** [comment]
it's a species, therefore, speciesm

**Comment by VampyrBytes** [comment]
@Fox-McCloud 
Can you add them to the *help section too please? Updated it not long ago and don't want to see it fall behind again

**Comment by Iamgoofball** [comment]
dressing up in an animal costume and going to conventions doesn't make you that animal and not human, your argument is void

regardless, what matters is that this goes against what the goons wanted done with their assets and code, confirming their fears that their sounds and assets would be used for furry shit.

**Comment by scrubmcnoob** [comment]
@Iamgoofball bro this is not real life, its a 2d spess man game people enjoy.
If people want to play a different race besides a boring human, then let them.
Oh no a wolf sound comes from this animal species, what shall I ever do...
But really, @Fox-McCloud shouldn't tangerines get this as well? Feel like it just negate any flame for leaving them out of the update.

**Comment by Iamgoofball** [comment]
why would a type of orange get to howl

**Comment by scrubmcnoob** [comment]
Its a pun off the name of Tarjaran, the other fur covered race in this video game.

**Comment by FlattestGuitar** [comment]
Waht... 
:+1:?

**Comment by CrAzYPiLoT-SS13** [comment]
:+1: Eh, why the hell not.

**Comment by Dumbdumn5** [comment]
Weird, but eh, how loud will it be, how long will it be, and good lord if it's as spammable as Vox shriek and is both long and loud, there may be something much worse upcoming. Still, fits along with the squishing we have for slime people, beeps we have for synthetics, shrieks we have for Vox, so on. Swear we need a sound emote for all of the races. Riot shield bashing effect on Kidan, anyone?

**Comment by Mijhead** [comment]
needs speshul snohflek lebbel :smirk:

**Comment by Vivalas** [comment]
>massive 10 second cooldown

**Comment by tigercat2000** [comment]
@Vivalas considering the *scream cooldown is 5 seconds

**Comment by FlattestGuitar** [comment]
Okay, only now has it occured to me that actually listening to the sounds might be a good idea.

Jesus Christ, they are terrible. The growls are awful and the howl lasts 6 seconds and is awful as well.

Let's get better sounds first, maybe? Like, really, they're terrible.

**Comment by Mijhead** [comment]
Yeah, I just listened to them as well. They're pretty horrible, I have to agree with @FlattestGuitar on that. I'd recommend finding some good sounds if you really want to have your vulpakanin's growl and howl. Just because Goon has some neat assets doesn't mean we need to grab all of them.

**Comment by Nienhaus** [comment]
>Howling
Hot

**Comment by TullyBurnalot** [comment]
I'll just say this:

This is literally what wolf howling sounds like.

As for the growls, eeeeh, they sound more like a snarl than a growl to me. Kinda expected them to be something like a low, rumbling "grrrrrrrrrrrrrrr".

**Comment by FlattestGuitar** [comment]
I'm not sure I support that being a "realistic howl". They're supposed to be a civilized species, not forest animals.

**Comment by TullyBurnalot** [comment]
True, but then you'd have to think up how a civilized wolf-people civilization would howl.

ALIEN BIOLOGY, GO!

**Comment by Mijhead** [comment]
That, and it's a low quality howl. You can hear the static.

**Comment by MactoPerFuror** [comment]
What the hell @Fox-McCloud ? You label @KasparoVy IPC thing with the special snowflake tag but not this? What is wrong with you? Is this more like you just attacking other peoples work or something? Why do you think this is not that? (honestly the label makes no sense either since its a social thing being used with it, i know its not being used for its code terminology)

This extreme hypocrisy is unreal, at this stage you must either be a charlatan or a fool.

**Comment by TullyBurnalot** [comment]
@Mijhead Maybe it's my sound system, but I can't hear static or sharp noises.

**Comment by tigercat2000** [comment]
@MactoPerFuror Attacking pull request authors in the comments is not allowed by our contributing guidelines. Especially over bad jokes.

**Comment by scrubmcnoob** [comment]
@MactoPerFuror Its literally a 'meme' tag. There is nothing more to it than that, it does not add to the content in anyway. Its just an online vidya game meme. Just chill.

**Comment by MactoPerFuror** [comment]
@tigercat2000 There is literally nothing here to tell me that this is a joke. @scrubmcnoob Shitposting and memes are now synonymous with opinions and criticism in this community, so I spose you speak the truth.

**Comment by Mijhead** [comment]
@tigercat2000 ,I'd like to point out, this doesn't seem to be a joke, atleast as far as I've heard from a headmin.

**Comment by Magikarpet** [comment]
The feature itself is alright, but as said by  @FlattestGuitar , this seems a bit silly due to the howling being specifically produced for feral creatures(werewolves in this case) while being meant to literally give you some sense of panic/fear, seeming incredibly odd and strange for a species meant to be civilized. This ignores also how weird and out of place they sound with the current ambiance/sounds of the game and the feeling it's supposed to output. Being toned down and made more appropriate for vulps(with the current sounds of the game) would make this much nicer(ignoring the hilarity behind how much is breaks IC emmulsions by listening to a non-feral species howl)

**Comment by tigercat2000** [comment]
@Mijhead @MactoPerFuror 
I was refering to the "special snowflake" tag; It's intended as a joke for pull requests that do nothing but aesthetic features, especially when related to typically "furry" things.

**Comment by Iamgoofball** [comment]
Then why isn't it on this PR?

**Comment by Mijhead** [comment]
Exactly what Goofball said.

**Comment by KasparoVy** [comment]
lmao you guys are too serious

**Comment by FlattestGuitar** [comment]
kek.

**Comment by TullyBurnalot** [comment]
I still say these aren't proper growls. These are snarls.

The pedant in me is crying.

**Comment by MactoPerFuror** [comment]
Why should we not be taking it serious? Why trivialize it? Is there only an issue if it is with you? I am actually disgusted by the amount of hypocrisy done by fox. I think this is even more self serving bullshit instead of serving the community.

There is no standards set for this kind of behavior and I take it as a serious issue. I am also disgusted by the allowance of it set by the management.

It seems the only distance set for when it has gone too far, is when someone points it out.

**Comment by Aurorablade** [comment]
@MactoPerFuror There is no hypocrisy, instead of trying to point our some conspiracy,bias or something other to stir up trouble in a guise of debate, though mind you are allowed A CIVIL opinion let us try this:

I want everyone in this comment section to do some positive feedback in such that instead of OMG THIS SOUND SUCKS NO MERGE THIS MEME FLUFFY STUFF (no not you @TullyBurnalot you coo) to do this, find a nice free quality sound you like better and link to it as a suggestion.

and instead of OMG BIAS TOWARD VULVAFLUFFYKANIN find us some NICE, FREE TO USE sounds for Cats (taraj), Lizards (unathi..bonus points if its angry sounding) and Insect (for kidan/crabs...REALLY bonus points for angry crab noise)

Can we all do that? Help us coders out! I am sure @Fox-McCloud would be MORE THEN HAPPY to use these sounds if someone helps us with links to quality ones.

All good? You have your homework, okay go!

**Comment by Mijhead** [comment]
@Aurorablade , that is a mainly stuff the PR maker should be doing. If he wants to have vulps howl and growl, he should have good sounds for them. It is not my job to do his work. As it stands now these sounds aren't just out of place on a station like Paradise, whilst they fit on Goon, but the howl also sounds like an incredibly old soundfile, and the growls are actually an earsore, with the relatively sharp and loud noise at the start of each soundfile, to the point that it physically hurts my ears to hear. And I am fairly certain I won't be the only one complaining in the event this goes live. And again, it is not my or anyone else's in the comment section job to find the soundfiles
 Its up to the PR-maker to deliver quality, and I believe hw has failed to do so in this case. Which, mightI add, isnt the first time in relation to.this specific race; on first implementation the lore for this race was plainly shit and snowflakey as hell, and I believe it deterred from the RP-quality it was supposed to give because of it. Not to mention how bland and unoriginal of an idea it is, with literally no game mechanics but language to back it up, unlike the other races.

**Comment by taukausanake** [comment]
Only because I want [this to happen](https://youtu.be/DsXkpSJOYyw?t=42s). Weee, pirates...

Yeah, these sounds aren't as good as they need to be, but they're from Goon like our new scream sounds. Those aren't as good either in my opinion but we got them all the same. We might as well start going with more "realistic" sounds as they come from Goon so as to not break the inconsistency.

And I'm not trying to be salty. I'm accepting this even though I want better sounds. Not likely to find any anyways so why complain besides the normal hate that's being spouted lately?

**Comment by Aurorablade** [comment]
@Mijhead i do not feel that the race has deterred from rp at all, but that is simply a contrast in opinion.
But i am slightly offended by your statement about the PR maker providing quality. Consider this, We work for free, We work with what we have. Consider a moment that we might not have the time or simply cannot find something another player might consider quality. I tried my best to find some cat noises for adding to tarj for this PR on google and my googlefu failed me.

There comes a time we need HELP from the community to improve on things or to give us CONSTRUCTIVE feed back with a possible solution on how it could be better. Though we have to take that feedback and deem if it is in fact feasible. And while we do accept feedback..we are also players, we like to add things that make other things fun for us.

While i have gotten conflicting opinions on my organ pr, some opinions have been at a point where i kick and scream, but you wanna know things i have added for a selfish reason just cuase i think its fun?
Slaughter Demons,
Honktumor,
and even if it was by request, the nyacromancer stone.
We highlight slaughter demons here cuase they can rip your six way to moonday and cause no end of salt.

But you know what? Its a game, we code to have fun or to improve things in a manner we feel is an improvement. Now i my next statment is going to be rude...and possibly unprofessional, for my peers i do apologize for it.

But i am sorry, if a pixel representation of a fox/wolf/canine in a video game and its sounds proposed for it is so bad that it is apprently sucking our your soul and giving you a brain tumor that you have to scream about it. Then we all need to take a step back.

**Comment by MactoPerFuror** [comment]
@Aurorablade So there is nothing to my claims just because you say otherwise. What on earth makes you think my claims are invalid and are just some guise? What have I said to imply that I am creating some ruse? Outrage cannot exist? I am not biting at the heels of vulpkanin's or furry shit. Also who are you to judge that and just brush it away for everyone?

**Comment by Mijhead** [comment]
Doesn't mean that we should just let low-quality happen because "its from goon" and "we already got some bad sounds from them so we might as wel keep doing it". If anything, the fact that we already got something arguably worse should be reason wenough to -not- port more bad sounds. It's generally the idea that PRs are supposed to improve, not negatively affect server quality, is it not?

**Comment by Aurorablade** [comment]
@Mijhead its not just cuase its from goon. But i am serious that if anyone has link to a better quality free use sound please post a link to it here.

**Comment by Mijhead** [comment]
The vibe I'm getting here right now is that because my critique doesn't supply any replacements, it shouldn't be considered. Have you considered that maybe not every pull made should be merged? Or atleast,not until it contains something that wouldn't be a detriment to the server, whether through addition of a commonly available feature which is of low quality, or any other? And about your quality comment and being offended by my statement; just take more time to ensure quality. This isn't a race. And I also disagrew with your implication that it is just an opinion of quality a coder might have issues with. There is an objective component to it as well.

**Comment by TullyBurnalot** [comment]
To expand on what @Mijhead said, I am not averse to the notion of sounds, but I'd much rather the Howl have its volume reduced (fucking please), and the growl sounds receive a uh... I dunno, better option?

I hate to beat a dead horse, but they sound much more like rabid snarls a wild dog would do rather than something a civilized sentient humanoid would produce from their vocal chords.

**Comment by Aurorablade** [comment]
@Mijhead well i do apologize as it was not my intent to give of that vibe. It should be considered, But also consider we may be having trouble finding an alternative that would be good. opinion on sound can be subjective. I mean i took two months with my organ PR and still got some bugs. I did mean the quality thing as just one person. Again on sound, now apply to music..while some people like say country music, I am not a huge fan of it.

Again sorry if i pull off that vibe toward you, I am just a person who likes help where help can be given. Regardless this will have to go under the same standards for merger as every other PR.

@TullyBurnalot if we do find another growl option..i would like to wild dog option avalible for most organic raises for when my human is off her pills so i can just growl at people....>.>

**Comment by scrubmcnoob** [comment]
I am going to go out on a limb here and say that you, @MactoPerFuror , have a predisposed biased against furies. For what ever the case is, I really don't care but what I do actually care about is sound quality because that is a legitimate complaint; rather than some silly trivial Github tag that will not be seen once this goes off Page 1.

Personally, I am not a stickler about certain sounds, and have not played goon a terrible ton, so to me it won't feel completely out of place. Considering we have an ear piercing Vox screech, I don't think a howl will bring the server to hell.

Are there better sound files out there?
Yeah probably.

Is the concept good?
It's a good concept, though a bit nicer sound files be better for this merge.

**Comment by Mijhead** [comment]
With your music example, @aurorablade , theres a difference between taste and quality. I can most certainly see the quality of music i dont enjoy, and I can also see the lack of it in music I do enjoy. What you described is taste. What i mean is quality.

**Comment by TullyBurnalot** [comment]
@Aurorablade Well, yeah, but Vulpkanin are supposed to be civilized sentient humanoids. It stands to reason that, due to their canine and vulpine physiology, they'd be capable of producing growls, snarls and howls much like a regular Earth canine or vulpine, but the way the growls are now feels very much "Rabid Dog" rather than "Sentient Canine".

I would not like to see the PR shut down and scrapped, as the concept itself, to me, is intriguing (full species unique sounds?! Wooo!). However, I would much rather this be put on hold until we can find a suitable replacement that a majority of players will enjoy hearing.

**Comment by tigercat2000** [comment]
I found the 100% perfect sound for Tajara making noises
https://www.freesound.org/people/VisualPony/sounds/269112/

**Comment by Dinarzad** [comment]
http://www.soundsnap.com/tags/alligator

Got your Unathi hiss sounds, specifically the last file.
Fuckin Gators, man

**Comment by ZomgPonies** [comment]
This thread has given up on life.

**Comment by FlattestGuitar** [comment]
[this](http://www.freesound.org/people/tomc1985/sounds/84656/) and [this](http://www.freesound.org/people/tomc1985/sounds/84654/) both sounds like good growls. Not too loud, not too long, not too agressive.

Also, way better than what goon has to offer in terms of growling.

**Comment by TheDZD** [comment]
The sounds sound fine, although I really don't think they really fit as a normal player emote; they seem better suited to wolves or werewolves.

And @Iamgoofball, you don't need to play white knight for the Goons. They are more than capable of fighting their own battles, if they dislike this PR, any of their coders/admins can comment or contact one of us privately and we will refrain from adding this out of respect for their wishes. I understand that you believe that this will have an affect on other servers' policy, and while that is not necessarily untrue, there's a little thing called a slippery slope fallacy that you might want to avoid falling into.

On another note, nice to see one of the subreddit's resident shitposters here, contributing absolutely nothing meaningful. If you want to actually contribute to this argument, try to avoid attacks on others' character, and stick to a simple "I don't like this because I am uncomfortable with the connotations of it." I would prefer that the discussion here stays reasonable, but if it does not and people continue along the line that they're going on, I will lock the PR comments.

**Comment by TheDZD** [comment]
Oh right I forgot.

**Comment by ZomgPonies** [comment]
@MactoPerFuror Stay the fuck out of our PRs. You've already made enough an ass of yourself on reddit, don't make us go through the process of booting you out of here also.

**Comment by Dumbdumn5** [comment]
Good fucking lord, it's a sound people! (But really, unless it was changed, the growl just doesn't sound right)

**Comment by KasparoVy** [comment]
I think it's time to delete some comments, there's forum-style posts going on here that's done nothing but spam this PR's board with little in terms of input

**Comment by Deanthelis** [comment]
When adding an emote-able sound, you have to take into consideration that people will group up and spam it, since cooldowns are irrelevant when there's 5+ people doing it.
Here's an example: http://puu.sh/nI6qI.ogg
Genuinely awful, on both my good equipment, and my terrible equipment.

For my other criticism of this PR, the Howl sound specifically has a 10 second cooldown and a 6 second duration, meaning you can make /more/ noise with it than other emotes that last a shorter amount of time in spite of their shorter cooldown.

Only a few sounds are louder than the howl - screaming and bike horns are the ones I can think of at the moment. All of them have a sort of static/buzzing that is inherent to ripped or improperly prepared sounds, but if I'm honest, most of our sounds are like that anyhow, so I don't see that as a valid complaint here.

Lastly, the tone of the emotes. No other species so far has... Well, in my opinion it's edgy, when compared to the relatively cuter squishing or beeping noises other species have; *buzz is probably the most hostile-sounding out of all of them and it's not even close to any of these growls - and if I might be pedantic here, none of the growl sound effects are growls, they are barks that trail into growls, so the PR itself isn't technically accurate.

**Comment by KasparoVy** [comment]
If the growl was any less aggro, I figure it'd be too jokey to be worthwhile

**Comment by Deanthelis** [comment]
The rest of the emote sounds are jokey besides *scream kind of.

**Comment by KasparoVy** [comment]
Why continue/why not continue the trend?

PR's still good by me, but if we can dig up some more suitable sounds then sweet. Otherwise I still figure it's iight.

What @TheDZD said is valid though, that "the sounds sound fine, ... they seem better suited to wolves or werewolves."

However, these are BIG wolf/whatever-people. How wussy would they sound, do you think?

**Comment by Deanthelis** [comment]
I don't think every species needs unique sounds at all, let alone ones that don't match the ones that do exist - and IPC ones aren't unique to IPCs, they're for all synthetics. As it is now, only Slimepeople have a snowflake sound, but they're also an extremely snowflakey species anyway; if we're going to have sound emotes, they should probably be applied to every species, lest anyone get the idea that a given coder has favoritism for a given species.

**Comment by KasparoVy** [comment]
Not everyone barks, I don't think

By that logic, we could apply the emotes to corgis or something and it'd be mint

**Comment by DavidRossii** [comment]
Bar of 'muh furries' arguement, the howl sounds so out of place and is way too long. But if we're going equality standards, what about Taj mrowling? What about Unathi hissing too? Ehh, this'll get merged, but I'm just glad there's muzzles around the place to put on the dogs.

**Comment by KasparoVy** [comment]
If better sounds turn up, and sounds of good enough quality are found for that which you mentioned, then sure

This PR isn't dismissive of those things, it's just adding one part of it all

How out of place sounding are Vox screams, slime squishes, or any of the other sounds we can make with our numerous species?

Howls in any form aren't very short, at least not as short as our standard screams. By nature, they are slightly more lengthy

> If people spam it, it just gives a reason to 'muzzle the dogs', so to say, anyway
> I'm just glad there's muzzles around the place to put on the dogs.
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

**Comment by Vivalas** [comment]
>UristMcVulpakanin howls!
>UristMcVulpakanin howls!
>UristMcVulpakanin howls!
>UristMcVulpakanin howls! Their voice sounds a bit strained!
>UristMcVulpakanin howls! They twist in agony as their throat hemorrhages awfully!
>UristMcVulpakanin siezes up and falls over limp, their eyes dead and lifeless 

Oh god please, this would make this PR.

**Comment by KasparoVy** [comment]
That's pretty jokes, actually. Something that could be added for all screams and a custom message for synthetics or something like >blows out their speakers
or something, but I think that's outside the scope here.

**Comment by Regen1** [comment]
:-1: 
The sounds doesn't fit and are really obnoxious, especially when spammed

Make a much more punishing system for emote spamming first

**Comment by NTSAM** [comment]
Gib people that use sound emotes three times in 20 seconds.

**Comment by Spacemanspark** [comment]
Or just mute people for a full minute or so, if they use sound emotes more than five times in a minute.

**Comment by KasparoVy** [comment]
Maybe some other time. I hope it gets added eventually in some shape or form.

**Pull request closed without merging**